Epam/Java Basics_UpSkill Lab #1/Algorithmization/Decomposition_using_methods/src/com/androsor/decomp/HomeTask16.java

```diff
@@ -0,0 +1,76 @@
+package com.androsor.decomp;
+
+/**
+ * Write a program that determines the sum of n - digit numbers containing only odd digits. Also determine how many
+ * even digits are in the found amount. To solve the problem, use decomposition.
+ */
+
+public class HomeTask16 {
+
+    public static void main(String [] args) {
+
+        int n = HomeTask1.dataInput(" Введите число N = ");
+        while (n < 0) {
+            System.out.println(" Число должно быть положительным!");
+            n = HomeTask1.dataInput(" Введите число N = ");
+        }
+
+        int sum = sumArray(arrayDigitCapacity(n));
+
+        System.out.printf(" Сумма нечетных цифр n-значного числа равна %d \n", sum);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.